pkg/standalone/common.go

```diff
@@ -44,9 +47,104 @@ func binaryFilePath(binaryDir string, binaryFilePrefix string) string {
 }
 
 func DefaultComponentsDirPath() string {
-	return path_filepath.Join(defaultDaprDirPath(), defaultComponentsDirName)
+	return path_filepath.Join(defaultDaprDirPath(), utils.DefaultComponentsDirName)
+}
+
+func DefaultResourcesDirPath() string {
+	return path_filepath.Join(defaultDaprDirPath(), utils.DefaultResourcesDirName)
+}
+
+// GetResourcesDir returns the path to the resources directory if it exists, otherwise it returns the path of components directory.
+// TODO: Remove this function and replace all its usage with above defined `DefaultResourcesDirPath` when `--components-path` flag is removed.
+func GetResourcesDir() string {
+	defaultResourcesDirPath := DefaultResourcesDirPath()
+	if _, err := os.Stat(defaultResourcesDirPath); os.IsNotExist(err) {
+		return DefaultComponentsDirPath()
+	}
+	return defaultResourcesDirPath
 }
 
 func DefaultConfigFilePath() string {
 	return path_filepath.Join(defaultDaprDirPath(), defaultConfigFileName)
 }
+
+// copyFilesAndCreateSymlink copies files from src to dest. It deletes the existing files in dest before copying from src.
+// this method also deletes the components dir and makes it as a symlink to resources directory.
+// please see this comment for more details:https://github.com/dapr/cli/pull/1149#issuecomment-1364424345
+// TODO: Remove this function when `--components-path` flag is removed.
+func copyFilesAndCreateSymlink(src, dest string) error {
+	var err error
+	if _, err = os.Stat(src); err != nil {
+		// if the src directory does not exist, create symlink and return nil, because there is nothing to copy from.
+		if os.IsNotExist(err) {
+			err = createSymLink(dest, src)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+		return fmt.Errorf("error reading directory %s: %w", src, err)
+	}
+	if err = moveDir(src, dest); err != nil {
+		return err
+	}
+	if err = createSymLink(dest, src); err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
Is this required here now? Can this be a part of `standalone.go` directly? This is no more a file abstraction, it's directly a bunch of operations now.